python/tk_multi_workfiles/actions/file_action_factory.py

```diff
@@ -36,158 +37,336 @@
 from ..file_item import FileItem
 from ..user_cache import g_user_cache
 
+
 class FileActionFactory(object):
     """
+    Builder for the file open menu. Based on the current selection, the build will enumerate a list of actions,
+    grouped actions and seperators that should be displayed in the UI.
     """
-    
-    def get_actions(self, file, work_area, file_model, workfiles_visible=True, publishes_visible=True):
+
+    def __init__(self, work_area, file_model, workfiles_visible=True, publishes_visible=True):
```

Comment:
Now that you're doing part of the work in `__init__`, it would imply that the Factory is less dynamic. Meaning that when the work area changes or the model changes, you have to create a new factory instance. I'm guessing that's fine and potentially gives a slight performance boost? I'm just curious about the change really since I'm not familiar with how the class is used in the app. :)